src/core/xds/grpc/xds_listener_parser.cc

```diff
@@ -264,13 +264,20 @@ XdsListenerResource::HttpConnectionManager HttpConnectionManagerParse(
           }
           continue;
         }
-        std::optional<XdsHttpFilterImpl::FilterConfig> filter_config =
-            filter_impl->GenerateFilterConfig(name, context,
-                                              std::move(*extension), errors);
-        if (filter_config.has_value()) {
-          http_connection_manager.http_filters.emplace_back(
-              XdsListenerResource::HttpConnectionManager::HttpFilter{
-                  std::string(name), std::move(*filter_config)});
+        std::optional<XdsHttpFilterImpl::XdsFilterConfig> filter_config =
```

Comment:
Is this a behavior change?

Reply:
No, it's not a behavior change.  What's happening here is that it's using both the old and new code paths for handling the filter configs, and storing the result of both in the parsed Listener object.  That way, the code in `XdsConfigSelector` can continue to support both code paths.  And it avoids breaking things on the server side, which doesn't yet support the new code-paths (that will be a separate set of work to update, not in the critical path for the initial ext_proc-on-client deliverable).